astropy/io/ascii/tests/test_c_reader.py

```diff
@@ -244,17 +244,19 @@ def test_conversion(parallel, read_basic):
     to strings.
     """
     text = """
-A B C D E
-1 a 3 4 5
-2. 1 9 10 -5.3e4
-4 2 -12 .4 six
+A B C D E F G
+1 a 3 4 5 6 7
+2. 1 9 10 8.7 6 -5.3e4
+4 2 -12 .4 - + six
```

Comment:
Which dash is this? Maybe a comment to clarify for future reference?

Reply:
I did not add any tests for en dash `–` or em dash `—` (GH code mode really seems to render -, – and — almost indistinguishable ;-), since they were already working correctly. Maybe should, but then there are loads of non-numerical characters that shall trigger `str` conversion.